cmd/ubuntu-image/main.go

```diff
@@ -2,41 +2,142 @@ package main
 
 import (
 	"fmt"
+	"io/ioutil"
 	"os"
 
 	"github.com/canonical/ubuntu-image/internal/commands"
+	"github.com/canonical/ubuntu-image/internal/helper"
+	"github.com/canonical/ubuntu-image/internal/statemachine"
 	"github.com/jessevdk/go-flags"
 )
 
+// helper variables for unit testing
 var osExit = os.Exit
+var captureStd = helper.CaptureStd
+var stateMachineInterface statemachine.SmInterface
+var imageType string = ""
 
 var stateMachineLongDesc = `Options for controlling the internal state machine.
 Other than -w, these options are mutually exclusive. When -u or -t is given,
 the state machine can be resumed later with -r, but -w must be given in that
-case since the state is saved in a .ubuntu-image.pck file in the working directory.`
+case since the state is saved in a .ubuntu-image.gob file in the working directory.`
+
+func executeStateMachine(commonOpts *commands.CommonOpts, stateMachineOpts *commands.StateMachineOpts, ubuntuImageCommand *commands.UbuntuImageCommand) {
```

Comment:
This is not a problem at all, just curiosity: are there any formatting guidelines for Go mentioning anything about line width, like we have in PEP8 for instance? I don't mind longer lines myself, since sometimes I think the 80 width limit from PEP8 is a bit too restrictive, especially on todays screens. Just wondering.

Reply:
Some of the go linters are quite happy to complain about such things.
I advised golangci-lint, but it's a bit of a pain to get running.
Maybe there is something else that is easier to integrate.

Reply:
I am actually using the official `golint` tool before making any commits/PRs. It's part of the `goreportcard` tool as well.